app/scripts/main.js

```diff
@@ -45,22 +47,72 @@ $(function () {
     return where;
   };
 
+  let changeData = () => {
+    $data.empty();
+    let where = createFilter();
+    let records = filter(pricingData, where);
+    let sortedRecords = sort(records);
+    for (let record of sortedRecords) {
+      $data.append('<tr><td>' + record.region + '</td><td>' + record.type + '</td><td>' + record.size + '</td><td>' + record.price + '</td></tr>');
+    }
+  }
+
   window.callback = function (data) {
     loadData(data);
     let uniqueValues = unique(pricingData, ['region', 'type', 'size']);
     render(uniqueValues);
+    $('select').on('change',changeData);
+  };
 
-    let $data = $('#pricing-data');
-    $('select').on('change', function () {
-      $data.empty();
-      let where = createFilter();
-      let records = filter(pricingData, where);
+  $sortIcon.on("click", () => {
+    if(!sortOrder) {
+      $sortIcon.removeClass('glyphicon-triangle-top');
+      $sortIcon.addClass('glyphicon-triangle-bottom');
+      sortOrder = true;
+    } else {
+      $sortIcon.removeClass('glyphicon-triangle-bottom');
+      $sortIcon.addClass('glyphicon-triangle-top');
+      sortOrder = false;
+    }
+    if($data.html()) {
+      changeData();
+    }
+  })
 
-      for (let record of records) {
-        $data.append('<tr><td>' + record.region + '</td><td>' + record.type + '</td><td>' + record.size + '</td><td>' + record.price + '</td></tr>');
+  let sort = records => {
+    let realData = [];
+    let imagData = [];
+    for(let record of records) {
+      if(record.price != "N/A*") {
+        realData.push(record);
+      } else {
+        imagData.push(record);
       }
-    });
-  };
+    }
+    if(!sortOrder) {
+      for(let i=0; i<realData.length - 1; i++) {
+        for(let j=0; j<realData.length - i -1; j++) {
+          if(parseFloat(realData[j].price) > parseFloat(realData[j+1].price)) {
+            let tmp = realData[j];
+            realData[j] = realData[j+1];
+            realData[j+1] = tmp;
+          }
+        }
+      }
+    } else {
+      for(let i=0; i<realData.length - 1; i++) {
+        for(let j=0; j<realData.length - i -1; j++) {
+          if(parseFloat(realData[j].price) < parseFloat(realData[j+1].price)) {
+            let tmp = realData[j];
+            realData[j] = realData[j+1];
+            realData[j+1] = tmp;
+          }
+        }
+      }
+    }
```

Comment:
This sorting logic could have been somewhat simplified. Find a way to use sort builtins i.e, `Array.sort`